pkg/webhook/clusterset/validatingWebhook.go

```diff
@@ -179,6 +180,11 @@ func (a *AdmissionHandler) validateUpdateRequest(request *v1.AdmissionRequest) *
 
 		// the cluster is claimed from clusterpool, do not allow to update clustersetlabel.
 		if clusterDeployment != nil && clusterDeployment.Spec.ClusterPoolRef != nil {
+			//For upgrade from 2.4 to 2.5, we need to move all clusters from empty set to "default" set.
+			if currentClusterSetName == defaultSetName {
```

Comment:
do we need to check if the old cluster has no clusterset here?

Reply:
done